spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -77,23 +77,32 @@ public DatastoreTemplate(Datastore datastore,
 	}
 
 	@Override
-	public <T> void save(T instance) {
-		this.datastore.put(convertToEntity(instance));
+	public <T> T save(T instance) {
+		Entity saved = this.datastore.put(convertToEntity(instance));
+		return this.datastoreEntityConverter.read((Class<T>) instance.getClass(), saved);
 	}
 
 	@Override
 	public <T> void deleteById(Object id, Class<T> entityClass) {
 		this.datastore.delete(getKeyFromId(id, entityClass));
 	}
 
+	@Override
+	public <T> void deleteAllById(Iterable<?> ids, Class<T> entityClass) {
+		List<Key> keys = getKeysFromIds(ids, entityClass);
+		this.datastore.delete(keys.toArray(new Key[keys.size()]));
+	}
+
 	@Override
 	public <T> void delete(T entity) {
```

Comment:
Can we return a `boolean` determining whether the delete occurred? Otherwise, we need to make sure that an exception is thrown when delete doesn't happen.

Reply:
we cannot. The delete function from the client lib is void and it says literally that deleting non-existing keys is "OK".

Reply:
We can't tell if something was deleted or not. though of course if it did exist and the delete failed we would have an exception.